x11-misc/sddm/sddm-0.20.0-r2.ebuild

```diff
@@ -0,0 +1,180 @@
+	touch 01gentoo.conf || die
+	touch sddm-02wayland-kwin.conf || die
+	touch sddm-02wayland-weston.conf
+
+cat <<-EOF >> 01gentoo.conf
+[General]
+# Remove qtvirtualkeyboard as InputMethod default
+InputMethod=
+EOF
+
+cat <<-EOF >> sddm-02wayland-kwin.conf
+#This file set SDDM to use wayland backend (kwin)
+
+[General]
+DisplayServer=wayland
+GreeterEnvironment=QT_WAYLAND_SHELL_INTEGRATION=layer-shell
+
+[Wayland]
+CompositorCommand=kwin_wayland --drm --no-lockscreen --no-global-shortcuts --locale1
+EOF
+
+cat <<-EOF >> sddm-02wayland-weston.conf
+# This file set SDDM to use wayland backend (weston)
+
+[General]
+DisplayServer=wayland
+
+[Wayland]
+CompositorCommand=weston --shell=fullscreen-shell.so
+EOF
+
+	cmake_src_prepare
+
+	if ! use test; then
+		sed -e "/^find_package/s/ Test//" -i CMakeLists.txt || die
+		cmake_comment_add_subdirectory test
+	fi
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_MAN_PAGES=ON
+		-DDBUS_CONFIG_FILENAME="org.freedesktop.sddm.conf"
+		-DRUNTIME_DIR=/run/sddm
+		-DSYSTEMD_TMPFILES_DIR="/usr/lib/tmpfiles.d"
+		-DNO_SYSTEMD=$(usex !systemd)
+		-DUSE_ELOGIND=$(usex elogind)
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+
+	insinto /etc/sddm.conf.d/
+	doins "${S}"/01gentoo.conf
+	if use wayland || use X; then
+		if use kde; then
+			newins "${S}"/sddm-02wayland-kwin.conf 02-wayland.conf
+		else
+			newins "${S}"/sddm-02wayland-weston.conf 02-wayland.conf
+		fi
+	fi
+}
+
+pkg_postinst() {
+	tmpfiles_process "${PN}.conf"
+
+	elog "NOTE: If SDDM startup appears to hang then entropy pool is too low."
+	elog "This can be fixed by configuring one of the following:"
+	elog "  - Enable CONFIG_RANDOM_TRUST_CPU in linux kernel"
+	elog "  - # emerge sys-apps/haveged && rc-update add haveged boot"
+	elog "  - # emerge sys-apps/rng-tools && rc-update add rngd boot"
+	elog
+	elog "SDDM example config can be shown with:"
+	elog "  ${EROOT}/usr/bin/sddm --example-config"
+	elog "Use ${EROOT}/etc/sddm.conf.d/ directory to override specific options."
+	elog
+	elog "For more information on how to configure SDDM, please visit the wiki:"
+	elog "  https://wiki.gentoo.org/wiki/SDDM"
+	if has_version x11-drivers/nvidia-drivers; then
+		elog
+		elog "  Nvidia GPU owners in particular should pay attention"
+		elog "  to the troubleshooting section."
+	fi
+	elos ""
+	elog "After testing is possible to be sure that wayland support work"
+	elog "But it still in a experimental phase"
+	elog "For Wayland support, you need tp enable the wayland useflag"
+	elog "An 02wayland.conf file will be created setting wayland as Display server"
+	elog "If kde useflag is also enable the compositor will be change from Weston to KWin"
+	elog ""
+	elog "If X and wayland useflag are both set, you need to manually configure sddm"
+	elog "It will use the default config"
+
+	optfeature "Weston DisplayServer support (EXPERIMENTAL)" dev-libs/weston
+	optfeature "KWin DisplayServer support (EXPERIMENTAL)" kde-plasma/kwin
```

Comment:
Since we are RDEPENDing on these packages, I feel the optfeature is unnecessary .